granulate_utils/metrics/yarn/utils.py

```diff
@@ -159,7 +161,13 @@ def detect_yarn_config(*, logger: Union[logging.Logger, logging.LoggerAdapter])
     if yarn_home_dir := find_yarn_home_dir(logger=logger):
         logger.debug(f"found YARN home dir: {yarn_home_dir}")
         yarn_site_xml_file = Path(yarn_home_dir).joinpath(RELATIVE_YARN_SITE_XML_PATH)
-        return read_config_file(yarn_site_xml_file, logger=logger)
+        if (conf := read_config_file(yarn_site_xml_file, logger=logger)) is not None:
```

Comment:
`read_config_file` API is not meant for such usage - it will log an error in this flow. I'm okay with removing the error log and adding an error log in this function (`detect_yarn_config`), something like "failed to detect config".